### PR TITLE
Update `Boolean` `transform` to avoid running a check for nans if the `dtype == bool`

### PR DESCRIPTION
**Pull request by ParthivNaresh**

### CODE REVIEW

**Comment by Cmancuso** [comment]
Do we need to test that Value Error? Otherwise lgtm

File: woodwork/logical_types.py [hunk 1]
```diff
@@ -223,15 +223,17 @@ def transform(self, series, null_invalid_values=False):
...
-                )
-            series.fillna(self.cast_nulls_as, inplace=True)
+        if not pdtypes.is_dtype_equal("bool", series.dtype):
+            if (is_dask and series.isna().any().compute()) or (
+                not is_dask and series.isna().any()
+            ):
+                if self.cast_nulls_as is None:
+                    raise ve
+                series.fillna(self.cast_nulls_as, inplace=True)
         series = _coerce_boolean(series, True)
```
Comment (bchen1116) [inline woodwork/logical_types.py:237]: Should this also be in the `if` statement? We shouldn't need to coerce if the type is 'bool'

**Comment by bchen1116** [comment]
Might be useful for us to run our own LG tests to make sure this is ok!

**Comment by ParthivNaresh** [comment]
@Cmancuso Added a test just to be sure!

**Review by bchen1116 — APPROVED** [review]
Nice!

**Pull request merged**